playbooks/robusta_playbooks/oom_killer.py

```diff
@@ -37,43 +55,211 @@ def oom_killer_enricher(event: NodeEvent):
-
-        for v in container_id_to_oom_killed_state.values():
-            status = v[0]
-            state = v[1]
-
-            dt = parse_kubernetes_datetime_to_ms(state.terminated.finishedAt)
-            oom_kills.append([dt, pod.metadata.name, status.name, status.image])
+    if not isinstance(event, PrometheusKubernetesAlert):
+        logging.info("OOMKillerEnricher can only be triggered by prometheus alerts")
+        return
 
-    oom_kills.sort(key=lambda o: o[0])
+    oom_kill_reason_investigator = KubernetesOomKillReasonInvestigator(node, event.alert, config)
+    oom_kills_extractor = OomKillsExtractor(config, node, oom_kill_reason_investigator)
+    oom_kills = oom_kills_extractor.extract_oom_kills()
 
-    if oom_kills:
+    if len(oom_kills) > 0:
         logging.info(f"found at least one oom killer on {node.metadata.name}")
+
+        oom_kills_headers = ["time", "pod", "container", "image", "memory_specs", "estimated_reason"]
+        oom_kills_rows = [
+            [oom_kill.time, oom_kill.pod_name, oom_kill.container_name,
+             oom_kill.image, repr(oom_kill.memory_specs), oom_kill.reason]
+            for oom_kill in oom_kills
+        ]
         event.add_enrichment(
             [
                 TableBlock(
-                    rows=oom_kills,
-                    headers=headers,
+                    rows=oom_kills_rows,
+                    headers=oom_kills_headers,
                     column_renderers={"time": RendererType.DATETIME},
                 )
             ]
         )
     else:
         logging.info(f"found no oom killers on {node.metadata.name}")
         event.add_enrichment([])
+
+
+@dataclass
+class MemorySpecs:
+    requests: Optional[str] = None
+    limits: Optional[str] = None
+
+
+@dataclass
+class OomKill:
+    time: float
+    pod_name: str
+    container_name: str
+    image: str
+    memory_specs: MemorySpecs
+    reason: Optional[str]
+
+
+# This class is used only because the OomKillsExtractor should appear before KubernetesOomKillReasonInvestigator, but still know its interface.
+class OomKillReasonInvestigator(metaclass=abc.ABCMeta):
+    @abstractmethod
+    def get_reason(self, oom_kill: OomKill) -> str:
+        return ""
+
+
+class OomKillsExtractor:
+    def __init__(self, config: OomKillerEnricherParams, node: Node, oom_kill_reason_investigator: OomKillReasonInvestigator):
+        self.config = config
+        self.node = node
+        self.oom_kill_reason_investigator = oom_kill_reason_investigator
+
+        self.memory_transformer = K8sMemoryTransformer()
+
+    def extract_oom_kills(self) -> List[OomKill]:
+        results: PodList = Pod.listPodForAllNamespaces(
+            field_selector=f"spec.nodeName={self.node.metadata.name}"
+        ).obj
+
+        oom_kills: List[OomKill] = []
+        for pod in results.items:
+            pod_oom_kills = self.get_oom_kills_from_pod(pod)
```

Comment:
might be cleaner to just use: 
for pod in results.items:
    oom_kills.extend(self.get_oom_kills_from_pod(pod))

Reply:
I agree, changed.